src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java

```diff
@@ -100,7 +115,7 @@ private void processQuery(JobContext context) throws IOException, InterruptedExc
     Type type = Objects.requireNonNull(bigQueryTable).getDefinition().getType();
 
     String query;
-    if (type == Type.VIEW || type == Type.MATERIALIZED_VIEW) {
+    if (type == Type.VIEW || type == Type.MATERIALIZED_VIEW || type == Type.EXTERNAL) {
```

Comment:
Is it easier if we just use BigQueryClient here to perform the query?

Reply:
The advantage of using the Hadoop client is that the table only gets materialized when Spark actually needs to read from it. For a large pipeline, several stages might run before this BQ source starts processing. This means that, if the pipeline fails in a previous stage, we now have to clean up a table that was never used.

Reply:
It also allows us to reuse the pattern we have been using so far

Reply:
Sounds good, let's keep this then